inc/class-mltools-gutenberg-config-generator.php

```diff
@@ -0,0 +1,478 @@
+			return;
+		}
+
+		$xpaths = $this->analyzeHtmlForXpaths( $block['innerHTML'] );
+
+		foreach ( $xpaths as $xpath ) {
+			$blockElement->addChild( 'xpath', $xpath );
+		}
+	}
+
+	/**
+	 * Analyzes HTML content to determine appropriate xpaths.
+	 *
+	 * @param string $html The HTML content.
+	 *
+	 * @return array Array of xpath expressions.
+	 */
+	private function analyzeHtmlForXpaths( $html ) {
+		$xpaths = [];
+
+		if ( empty( trim( $html ) ) ) {
+			return $xpaths;
+		}
+
+		$doc = new DOMDocument();
+		@$doc->loadHTML( '<div>' . $html . '</div>', LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD );
+
+		$xpath = new DOMXPath( $doc );
+
+		$textNodes = $this->findTextNodes( $doc->documentElement );
+
+		foreach ( $textNodes as $node ) {
+			if ( $node->nodeType === XML_TEXT_NODE ) {
+				$text = trim( $node->nodeValue );
+				if ( ! empty( $text ) ) {
+					$parent = $node->parentNode;
+					if ( $parent !== null ) {
+						$xpathExpression = $this->generateUniqueXPath( $parent );
+						if ( ! empty( $xpathExpression ) ) {
+							$xpaths[] = $xpathExpression;
+						}
+					}
+				}
+			}
+		}
+
+		$links = $xpath->query( '//a[@href]' );
+		foreach ( $links as $link ) {
+			$xpaths[] = $this->generateUniqueXPath( $link ) . '/@href';
+		}
+
+		$attrMap = [
+			'img'      => [ 'alt', 'title' ],
+			'input'    => [ 'placeholder', 'value' ],
+			'textarea' => [ 'placeholder' ],
+			'button'   => [ 'value' ],
+			'meta'     => [ 'content' ],
+		];
+
+		foreach ( $attrMap as $tag => $attributes ) {
+			$elements = $doc->getElementsByTagName( $tag );
+			foreach ( $elements as $element ) {
+				foreach ( $attributes as $attribute ) {
+					if ( $element->hasAttribute( $attribute ) && ! empty( trim( $element->getAttribute( $attribute ) ) ) ) {
+						$xpaths[] = $this->generateUniqueXPath( $element ) . '/@' . $attribute;
+					}
+				}
+			}
+		}
+
+		return array_unique( $xpaths );
+	}
+
+	/**
+	 * Recursively finds text nodes within an element.
+	 *
+	 * @param DOMNode $node The node to check.
+	 *
+	 * @return array Array of text nodes.
+	 */
+	private function findTextNodes( $node ) {
+		$textNodes = [];
+
+		if ( $node->nodeType === XML_TEXT_NODE ) {
+			$text = trim( $node->nodeValue );
+			if ( ! empty( $text ) ) {
+				$textNodes[] = $node;
+			}
+		}
+
+		if ( $node->hasChildNodes() ) {
+			foreach ( $node->childNodes as $child ) {
+				$textNodes = array_merge( $textNodes, $this->findTextNodes( $child ) );
+			}
+		}
+
+		return $textNodes;
+	}
+
+	/**
+	 * Generates a unique XPath for a DOM element.
+	 *
+	 * @param DOMElement $element The DOM element.
+	 *
+	 * @return string The XPath expression.
+	 */
+	private function generateUniqueXPath( $element ) {
+		if ( $element->nodeType !== XML_ELEMENT_NODE ) {
+			return '';
+		}
+
+		$nodePath = $element->nodeName;
+
+		if ( $element->hasAttribute( 'class' ) ) {
+			$classes         = explode( ' ', $element->getAttribute( 'class' ) );
+			$filteredClasses = array_filter(
```

Comment:
```suggestion
			$filteredClasses = array_filter( $classes, 'trim' );
```

I presume you will get the same result with that one-liner.